src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,977 @@
+
+			m_isPointerOver = false;
+			m_isPressed = false;
+
+			handled = args.Handled;
+
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerExited - handled=%d, parentIsSubMenu=%d.", this, handled, parentIsSubMenu));
+#endif // CMH_DEBUG
+
+			if (m_delayOpenMenuTimer != null)
+			{
+#if CMH_DEBUG
+        (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerExited - Stopping m_delayOpenMenuTimer.", this));
+#endif // CMH_DEBUG
+
+				m_delayOpenMenuTimer.Stop();
+			}
+
+			var ownerAsUIE = m_wpOwner?.Target as FrameworkElement;
+
+			if (!handled && ownerAsUIE != null && ownerAsUIE.IsLoaded)
+			{
+				Pointer pointer;
+				var pointerDeviceType = PointerDeviceType.Touch;
+
+				pointer = (args.Pointer);
+				pointerDeviceType = (pointer.PointerDeviceType);
+
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerExited - pointerDeviceType=%d.", this, pointerDeviceType));
+#endif // CMH_DEBUG
+
+				if (PointerDeviceType.Mouse == pointerDeviceType && !parentIsSubMenu)
+				{
+					UIElement subMenuPresenterAsUIE = m_wpSubMenuPresenter?.Target as UIElement;
+
+					if (subMenuPresenterAsUIE != null)
+					{
+						bool isOwnerOrSubMenuHit = false;
+						Point point = default;
+						PointerPoint pointerPoint;
+
+						pointerPoint = args.GetCurrentPoint(null);
+						point = (pointerPoint.Position);
+
+						var elements = VisualTreeHelper.FindElementsInHostCoordinates(point, ownerAsUIE, true /* includeAllElements */);
+
+						foreach (var element in elements)
+						{
+							if (ownerAsUIE == element || subMenuPresenterAsUIE == element)
+							{
+								isOwnerOrSubMenuHit = true;
+								break;
+							}
+						}
+
+						if (!isOwnerOrSubMenuHit)
+						{
+							elements = VisualTreeHelper.FindElementsInHostCoordinates(point, subMenuPresenterAsUIE, true /* includeAllElements */);
+
+							foreach (var element in elements)
+							{
+								if (ownerAsUIE == element || subMenuPresenterAsUIE == element)
+								{
+									isOwnerOrSubMenuHit = true;
+									break;
+								}
+							}
+						}
+
+						// To close the sub menu, the pointer must be outside of the opened chain of sub-menus.
+						if (!isOwnerOrSubMenuHit)
+						{
+							DelayCloseSubMenu();
+							args.Handled = true;
+						}
+					}
+				}
+
+				UpdateOwnerVisualState();
+			}
+		}
+
+		// PointerPressed event handler ensures that we're in the pressed state.
+		internal void OnPointerPressed(PointerRoutedEventArgs args)
+		{
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerPressed.", this));
+#endif // CMH_DEBUG
+
+			m_isPressed = true;
+
+			args.Handled = true;
+		}
+
+		// PointerReleased event handler shows the sub menu in the
+		// case of touch input.
+		internal void OnPointerReleased(PointerRoutedEventArgs args)
+		{
+			Pointer pointer;
+			PointerDeviceType pointerDeviceType = PointerDeviceType.Touch;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'pointerDeviceType'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)